File: TransCoder_tokenized_test_set_functions/cpp/FIND_MAXIMUM_DOT_PRODUCT_TWO_ARRAYS_INSERTION_0S.cpp
long long int MaxDotProduct ( int A [ ], int B [ ], int m, int n ) {
  long long int dp [ n + 1 ] [ m + 1 ];
  memset ( dp, 0, sizeof ( dp ) );
  for ( int i = 1;
  i <= n;
  i ++ ) for ( int j = i;
  j <= m;
  j ++ ) dp [ i ] [ j ] = max ( ( dp [ i - 1 ] [ j - 1 ] + ( A [ j - 1 ] * B [ i - 1 ] ) ), dp [ i ] [ j - 1 ] );
  return dp [ n ] [ m ];
}

File: TransCoder_tokenized_test_set_functions/java/FIND_MAXIMUM_DOT_PRODUCT_TWO_ARRAYS_INSERTION_0S.java
static int MaxDotProduct ( int A [ ], int B [ ], int m, int n ) {
  int dp [ ] [ ] = new int [ n + 1 ] [ m + 1 ];
  for ( int [ ] row : dp ) Arrays . fill ( row, 0 );
  for ( int i = 1;
  i <= n;
  i ++ ) for ( int j = i;
  j <= m;
  j ++ ) dp [ i ] [ j ] = Math . max ( ( dp [ i - 1 ] [ j - 1 ] + ( A [ j - 1 ] * B [ i - 1 ] ) ), dp [ i ] [ j - 1 ] );
  return dp [ n ] [ m ];
}